src/coreclr/tools/aot/ILCompiler.RyuJit/JitInterface/CorInfoImpl.Llvm.cs

```diff
@@ -83,7 +83,7 @@ public static int getSignatureForMethodSymbol(IntPtr thisHandle, IntPtr symbolHa
             if (node is IMethodNode { Offset: 0, Method: MethodDesc method })
             {
                 _this.Get_CORINFO_SIG_INFO(method, pSig, scope: null);
-                if (method.IsUnmanagedCallersOnly)
+                if (method.IsUnmanagedCallersOnly || node is RuntimeImportMethodNode)
```

Comment:
`CORINFO_CALLCONV_UNMANAGED` implies GC transition. Are we compensating for it somewhere?

Reply:
> Are we compensating for it somewhere?

There is no need as the API is only used to retrieve the signature from a function pointer representative. to construct the LLVM equivalent. Making it work for other cases would require adding a way to describe the "FCall calling convention", i. e. native ABI + no marshaling, no transition and no poll.

(Additionally, whether the transition will be needed is influenced by SPGCT and that is callsite specific.)